services/github/github-package-registry-downloads.service.js

```diff
@@ -0,0 +1,295 @@
+'use strict'
+
+const gql = require('graphql-tag')
+const Joi = require('@hapi/joi')
+const { downloadCount } = require('../color-formatters')
+const { metric } = require('../text-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { GithubAuthV4Service } = require('./github-auth-service')
+const { documentation, transformErrors } = require('./github-helpers')
+const { NotFound } = require('..')
+
+// https://developer.github.com/v4/object/registrypackagestatistics/
+// https://developer.github.com/v4/object/registrypackageversionstatistics/
+const packageStatistics = Joi.object({
+  statistics: Joi.object({
+    downloadsToday: nonNegativeInteger,
+    downloadsThisWeek: nonNegativeInteger,
+    downloadsThisMonth: nonNegativeInteger,
+    downloadsThisYear: nonNegativeInteger,
+    downloadsTotalCount: nonNegativeInteger,
+  }).required(),
+})
+
+const totalDownloadsSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      registryPackages: Joi.object({
+        nodes: Joi.array()
+          .items(packageStatistics)
+          .min(0)
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+const versionDownloadsSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      registryPackages: Joi.object({
+        nodes: Joi.array()
+          .items(
+            Joi.object({
+              version: packageStatistics.allow(null),
```

Comment:
This is `null` when a non-existent value is specified for `version` in the query